ci-operator/step-registry/aws/provision/kms-key/aws-provision-kms-key-ref.yaml

```diff
@@ -11,7 +11,10 @@ ref:
     requests:
       cpu: 10m
       memory: 100Mi
+  env:
+  - name: USE_DIFFERENT_AWS_KMS_KEY_FOR_COMPUTE
```

Comment:
I checked similar case (`azure-provision-disk-encryption-set`) on Azure platform, the idea is something like using `ENABLE_AWS_KMS_KEY_CONTROL_PLANE`, `ENABLE_AWS_KMS_KEY_COMPUTE`, and `ENABLE_AWS_KMS_KEY_DEFAULT_MACHINE` to control which kms should be provisioned. Once any env var is enabled, always provision a new kms key. Once that, we do not have to introduce `USE_DIFFERENT_AWS_KMS_KEY_FOR_COMPUTE`, and even do not need to add more installation job to cover, existing regular ci jobs can meet our coverage. WDYT.

also cc @jinyunma for comments if I am wrong.

Reply:
yes, exactly. On azure, we have a pair of e2e jobs to run those scenarios:
*  azure-ipi-des-mini-perm-arm-f14  
      ENABLE_DES_DEFAULT_MACHINE: "true"
-----> same des used on master and worker, also enable des configuration in default storage account
* azure-ipi-des-mini-perm-fips-amd-f28-destructive  
      ENABLE_DES_COMPUTE: "true"
      ENABLE_DES_CONTROL_PLANE: "true"
 -----> different des used on master and worker

Reply:
Updated.
`aws-ipi-byo-kms-etcd-encryption-arm-f7` uses default values (default platform).
`aws-ipi-byo-kms-etcd-encryption-fips-amd-f28-destructive` uses two different keys.